packages/discovery-react-components/src/components/CIDocument/components/CIDocumentContent/CIDocumentContent.tsx

```diff
@@ -134,27 +133,21 @@ const CIDocumentContent: FC<CIDocumentContentProps> = ({
   );
 };
 
-function highlightStyling(
-  highlightedLocationIdsByColor: { color: string; highlightLocationIds: string[] }[]
-): string[] {
-  return highlightedLocationIdsByColor.map(highlightData => {
-    const { color, highlightLocationIds } = highlightData;
-
-    return createStyleRules(highlightLocationIds, [
-      backgroundColorRule(color),
-      // Set z-index to -1 in order to push non-active fields back
-      zIndexRule(-1)
-    ]);
-  });
-}
-
 function createStyleRules(idList: string[], rules: string[]): string {
   return idList
     .map(id => `.${baseClassName} .field[data-field-id="${id}"] > *`)
     .join(',')
     .concat(`{${rules.join(';')}}`);
 }
 
+function highlightColoringFullArray(combinedHighlightsWithMeta: HighlightWithMeta[]) {
+  return combinedHighlightsWithMeta.map(highlightWithMeta => {
+    const locationId = getHighlightLocationId(highlightWithMeta);
+    const rules = `.${baseClassName} .field[data-field-id="${locationId}"] > * {background-color: ${highlightWithMeta.color}; z-index: -1;}`;
```

Comment:
I moved away from `createStyleRules` because it expects a much simpler shape, basically just a list of location ids and some rules that apply to all of them. We kinda made it work when I added just unique colors, but it's making less sense to continue using it as we're trying to add more context around these highlights. Personally, I'd prefer to stop using it instead of bending over backward to make our new data fit.

Reply:
I did leave the `createStyleRules` in place for the original handling of highlights, since I think we're still using the simpler version somewhere